samples/_emscripten/_audio/EmscriptenAudio/Voice.cpp

```diff
@@ -0,0 +1,345 @@
+{
+	const auto &bus = mBusses.at( busId );
+
+	if( bus.mPan )
+		return bus.mPan;
+	else
+		return bus.mGain;
+}
+
+// ----------------------------------------------------------------------------------------------------
+// Voice
+// ----------------------------------------------------------------------------------------------------
+
+VoiceRef Voice::create( const CallbackProcessorFn &callbackFn, const Options &options )
+{
+	VoiceRef result( new VoiceCallbackProcessor( callbackFn, options ) );
+	MixerImpl::get()->addVoice( result, options );
+
+	return result;
+}
+
+bool Voice::isDoneLoading(){
+	return mDoneParsing;
+}
+
+void Voice::create( const SourceFileRef &sourceFile, std::function<void(VoiceSamplePlayerNodeRef)> cb, const Options &options){
+	new VoiceSamplePlayerNode( sourceFile, options , cb );
+}
+VoiceSamplePlayerNodeRef Voice::create( const SourceFileRef &sourceFile, const Options &options )
+{
+	VoiceSamplePlayerNodeRef result( new VoiceSamplePlayerNode( sourceFile, options ) );
+	MixerImpl::get()->addVoice( result, options );
+
+	return result;
+}
+
+Voice::~Voice()
+{
+	MixerImpl::get()->removeVoice( mBusId );
+}
+
+void Voice::clearBufferCache()
+{
+	MixerImpl::get()->clearBufferCache();
+}
+
+float Voice::getVolume() const
+{
+	return MixerImpl::get()->getBusVolume( mBusId );
+}
+
+float Voice::getPan() const
+{
+	return MixerImpl::get()->getBusPan( mBusId );
+}
+
+void Voice::setVolume( float volume )
+{
+	MixerImpl::get()->setBusVolume( mBusId, volume );
+}
+
+void Voice::setPan( float pan )
+{
+	MixerImpl::get()->setBusPan( mBusId, pan );
+}
+
+void Voice::start()
+{
+	getInputNode()->enable();
+}
+
+void Voice::pause()
+{
+	getInputNode()->disable();
+}
+
+void Voice::stop()
+{
+	getInputNode()->disable();
+}
+
+bool Voice::isPlaying() const
+{
+	return getInputNode()->isEnabled();
+}
+
+NodeRef Voice::getOutputNode() const
+{
+	return MixerImpl::get()->getOutputNode( mBusId );
+}
+
+// ----------------------------------------------------------------------------------------------------
+// VoiceSamplePlayerNode
+// ----------------------------------------------------------------------------------------------------
+
+
+VoiceSamplePlayerNode::VoiceSamplePlayerNode( const SourceFileRef &sourceFile, const Options &options, std::function<void(VoiceSamplePlayerNodeRef)> cb )
+	: Voice()
+{
+
+
+	#ifndef CINDER_EMSCRIPTEN
+		size_t requiredSampleRate = audio::master()->getSampleRate();
+		SourceFileRef sf = requiredSampleRate == sourceFile->getSampleRate() ? sourceFile : sourceFile->cloneWithSampleRate( requiredSampleRate );
+
+		if( sf->getNumFrames() <= options.getMaxFramesForBufferPlayback() ) 
+		{
+			BufferRef buffer = MixerImpl::get()->loadBuffer( sf );
+			mNode = Context::master()->makeNode( new BufferPlayerNode( buffer ) );
+		} 
+		else {
+			mNode = Context::master()->makeNode( new FilePlayerNode( sf ) );
+		}
+	#else
+
+
+		auto loader = (WebAudioLoader*)sourceFile.get();
```

Comment:
Some feedback on this section of the code: if `WebAudioLoader` is a subclass of `ci::audio::SourceFile`, maybe it should be the thing we were discussing and calling `SourceFileWeb` (or maybe `SourceFileWebAudio` is a better fit, I don't know).

Reply:
Looking more into this you're right of course, it wasn't quite clear when first doing this how `ci::audio::Source` was working but taking a second look things seemed to make more sense. 
I've added a `SourceFileWebAudio` class to try and better match the other platforms.